dbt_copilot_helper/utils/validation.py

```diff
@@ -18,10 +23,82 @@ def validator(string):
     return validator
 
 
-def validate_s3_bucket_name(name):
+S3_BUCKET_NAME_ERROR_TEMPLATE = "Bucket name '{}' is invalid:\n{}"
+AVAILABILITY_UNCERTAIN_TEMPLATE = (
+    "Warning: Could not determine the availability of bucket name '{}'."
+)
+BUCKET_NAME_IN_USE_TEMPLATE = "Bucket name '{}' is already in use."
+
+
+def s3_bucket_name_is_available(name: str):
+    """
+    We try to find the bucket name in AWS.
+
+    The validation logic is:
+    True: if the response is a 200 (it exists and you have access - this bucket has probably already been deployed)
```

Comment:
Will this work cross account? Thinking your current session is the non-prod account but you have deployed an S3 bucket into the prod account.

Reply:
It won't, but it's still better to accept here, as there's no way to tell if it is the bucket for your deployment or for another account. It will have to fail on deployment in that case, which is better than stopping people rebuilding addons if they have already deployed their bucket once.

Reply:
My point with this is that if the bucket already exists and is in the prod account, the call will get a 40x, therefore raising a validation error.

Reply:
Oh, I see. I did have this check just throw up a warning rather than failing the check. I'll revert to that then I think. Unless you can see a better solution?

Reply:
Pre-deployment I don't see a better way, post deployment the `application.environments` bit could be used to check each env account, but that just feels like overkill. I think just a warning saying that the bucket may be taken is enough, and indicate that it may be a problem, but check your aws accounts.